File: code/method_block_binding.rb
class Action
  def self.say(&sentence)
    @name = 'Ann'
    p sentence.call
  end
end

class Person
  def initialize(name)
    @name = name
  end

  def greet
    Action.say { "My name is #{@name}!" }
    # "My name is Alex!"
  end
end

Person.new('Alex').greet

File: code/method_block_binding.py
class Action:
  name = 'Ann'

  @staticmethod
  def say(sentence):
    print(sentence())


class Person:
  def __init__(self, name):
    self.name = name

  def greet(self):
    Action.say(lambda: "My name is %s!" % self.name)


Person('Alex').greet()
